### ISSUE TITLE
Workspace - User can save the wrong rate

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#19685]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Issue found when executing PR** https://github.com/Expensify/App/pull/18454

## Action Performed:
1. Go to URL https://staging.new.expensify.com/
2. Login with any account
3. Go to Settings > Workspace > choose or create a workspace >Reimburse expenses
4. Change the rate and the unit
5. Check that you can't save a wrong rate: 1. or 0.

## Expected Result:
User can't save a wrong rate.

## Actual Result:
User can save the wrong rate.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.19.1

**Reproducible in staging?:** Yes

**Reproducible in production?:** new feature

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://platform.applause.com/services/links/v1/external/0468af9dd4ab6d9c43ab952bf3538f97905dcc92d1b02aed1627473ed50852f4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by trjExpensify** [comment]
👋 @youssef-lr @eVoloshchak @ShogunFire @joelbettner looks like a regression from [this PR](https://github.com/Expensify/App/pull/18454), is it on your radar to fix as this is only on staging right now?

**Comment by ShogunFire** [comment]
@trjExpensify I can make the correction right now but I don't know the process for regressions. Do I just need to do another PR ?

**Comment by trjExpensify** [comment]
Yup, let's do another PR right away. @youssef-lr @joelbettner @eVoloshchak if you guys agree this is a deploy blocker, let's add the label too.

**Comment by eVoloshchak** [comment]
Huh, I remember testing this case specifically and thinking `1.` or `7.` is a valid input and this is effectively the same as calling `.trim()` for a string. We've got all the information we need from the user, we should interfere as little as possible. 
And that's how it was implemented previously (currently on prod).
But I could see how it can be considered invalid, this might be an indication that the user mistyped the rate, so let's spin up PR with the fix

@trjExpensify, I don't know what qualifies to be a deploy blocker (is there a doc?) but this seems very minor to be a deploy blocker
